components/location/popup.jsx

```diff
@@ -0,0 +1,319 @@
+  );
+};
+
+const OpenHours = ({ closingHours, id, openingHours }) => {
+  if (openingHours === "never" || openingHours === "not today") {
+    return <PopupSection content={openingHours} id={id} />;
+  } else {
+    const openHours = closingHours.map((closeHour, i) => {
+      <p>
+        <span>{openingHours[i]}</span> to <span>{closingHours[i]}</span>
+      </p>;
+    });
+    return <PopupSection content={openHours} id={id} />;
+  }
+};
+
+const NoIdNeeded = ({ location }) => {
+  const { getTranslation } = useContext(LanguageContext);
+
+  if (location.someInfoRequired === true) {
+    return (
+      <span
+        className="noIdNeeded card-badge"
+        // data-translation-id="some_info_required"
+      >
+        {getTranslation("some_info_required")}
+      </span>
+    );
+  } else {
+    return (
+      <span
+        className="noIdNeeded card-badge"
+        // data-translation-id="no_id_needed"
+      >
+        {getTranslation("no_id_needed")}
+      </span>
+    );
+  }
+};
+
+const WarmingSite = () => {
+  const { getTranslation } = useContext(LanguageContext);
+  return (
+    <span
+      // Kanad note: this translation ID is wrong!
+      // data-translation-id="warming_site"
+      className="warmingSite card-badge"
+    >
+      {getTranslation("warm_up_here")}
+    </span>
+  );
+};
+
+// const Address = ({ address }) => {
+//   const googleMapDirections = `https://maps.google.com?saddr=Current+Location&daddr=${encodeURI(
+//     address
+//   )}`;
+//   return (
+//     <address>
+//       <a
+//         href={googleMapDirections}
+//         target="_blank"
+//         onClick="captureOutboundLink('${googleMapDirections}', 'directions')"
+//         rel="noreferrer"
+//       >
+//         {address}
+//       </a>
+//     </address>
+//   );
+// };
+
+const PublicTransit = ({ location }) => {
+  const options = location.publicTransitOptions;
+  const routes = options.map((option, i) => (
+    <React.Fragment key={`${option.routeName}-${i}`}>
+      {/** Kanad Note: I had to change all these span tags to div tags
+       * since it was causing issues when doing a ReactDOMServer render.
+       *
+       * I also got rid of the separate div with alt text and
+       * just added it as an aria-label below.
+       * Source: https://stackoverflow.com/a/38438002 */}
+      <div
+        aria-label={option.altText}
+        className="transit-route-option"
+        role="text"
+      >
+        <i className="material-icons-round route-icon">{option.icon}</i>
+        <h5
+          className="transit-route route-text"
+          style={{ backgroundColor: option.backgroundColor }}
+        >
+          {option.routeName}
+        </h5>
+        <h5 className="route-text">{option.distance}</h5>
+      </div>
+    </React.Fragment>
+  ));
+  return <PopupSection content={routes} id="public_transit" />;
+};
+
+// builds the section within the popup and replaces and URLs with links
+const SectionUrl = ({ id, value }) => {
```

Comment:
Possibly a better name for this? (I know it was just carried over from previous)

Reply:
Gonna leave this as is for now unless you have a better idea, but agreed that it's not named super well!